src/lib/caching/stores/GuildChannelInviteStore.ts

```diff
@@ -0,0 +1,130 @@
+import { RequestOptions, Routes } from '@klasa/rest';
+import { ProxyCache } from '@klasa/cache';
+
+import type { Client } from '../../client/Client';
+import type { Invite } from '../structures/Invite';
+import type { GuildChannel } from '../structures/channels/GuildChannel';
+import type { APIInviteData, InviteTargetUserType } from '@klasa/dapi-types';
+import type { Channel } from '../structures/channels/Channel';
+import type { TextChannel } from '../structures/channels/TextChannel';
+/**
+ * The store for {@link Invite guild invites} the channel has.
+ * @since 0.0.3
+ */
+export class GuildChannelInviteStore extends ProxyCache<string, Invite> {
+
+	/**
+	 * The {@link Client client} this store belongs to.
+	 * @since 0.0.3
+	 */
+	public readonly client: Client;
+
+	/**
+	 * The {@link Channel channel} this store belongs to.
+	 * @since 0.0.3
+	 */
+	public readonly channel: Channel;
+
+	/**
+	 * Builds the store.
+	 * @since 0.0.3
+	 * @param channel The {@link GuildChannel guild channel} this store belongs to.
+	 */
+	public constructor(channel: GuildChannel, keys: string[]) {
+		super(channel.client.invites, keys);
```

Comment:
Does the client have an invite store? Looking at https://github.com/dirigeants/core/pull/264/files#diff-0bd9f9e02a70d7079d024a9639819746R60, it's guilds that have the store, not the client; and if both have the store, then you should still probably use the guild's one and not the client's one

Reply:
Well kyra said to use the client store instead so idk

Reply:
CC @kyranet

Reply:
Kyra was correct, should be client.invites, guilds have them too, and fetch on each does different things based on how discords own routes work.

Reply:
alright.